internal/commands/vaultsecrets/run/run.go

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package run
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/hashicorp/consul-template/child"
+	preview_secret_service "github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/preview/2023-11-28/client/secret_service"
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/stable/2023-06-13/client/secret_service"
+	"github.com/hashicorp/hcp/internal/commands/vaultsecrets/secrets/appname"
+	"github.com/hashicorp/hcp/internal/pkg/cmd"
+	"github.com/hashicorp/hcp/internal/pkg/flagvalue"
+	"github.com/hashicorp/hcp/internal/pkg/format"
+	"github.com/hashicorp/hcp/internal/pkg/heredoc"
+	"github.com/hashicorp/hcp/internal/pkg/iostreams"
+	"github.com/hashicorp/hcp/internal/pkg/profile"
+)
+
+type RunOpts struct {
+	Ctx     context.Context
+	Profile *profile.Profile
+	IO      iostreams.IOStreams
+	Output  *format.Outputter
+
+	App           string
+	Command       string
+	PreviewClient preview_secret_service.ClientService
+	Client        secret_service.ClientService
+}
+
+func NewCmdRun(ctx *cmd.Context, runF func(*RunOpts) error) *cmd.Command {
+	opts := &RunOpts{
+		Ctx:           ctx.ShutdownCtx,
+		Profile:       ctx.Profile,
+		IO:            ctx.IO,
+		Output:        ctx.Output,
+		PreviewClient: preview_secret_service.New(ctx.HCP, nil),
+		Client:        secret_service.New(ctx.HCP, nil),
+	}
+
+	cmd := &cmd.Command{
+		Name:      "run",
+		ShortHelp: "Run a process with secrets from a Vault Secrets app.",
+		LongHelp: heredoc.New(ctx.IO).Must(`
+		The {{ template "mdCodeOrBold" "hcp vault-secrets run" }} command lets you
```

Comment:
Should this be `hcp vault-secrets apps run` given you are running a command for all the secrets of an app? This feels like a sub-action of apps?

Reply:
I think we should stick with top level for now as we don't know what enhancements are planned in future. Moving it under apps also locks us to one app - a design constraint we don't necessarily have to hold. We should run this through product if we don't anticipate anything more in future.

Reply:
Chatted on this point with the HVS tech lead and PM as well. Sounds like folks are in favor of keeping it top-level for now.